tox.ini

```diff
@@ -34,3 +34,12 @@ deps = mypy==1.10.0
 # mypy gives different results if you actually install the stuff before you check it
 # separate cache to stop weirdness around sharing cache with other instances of mypy
 commands = mypy --cache-dir="{envdir}/mypy_cache" --config-file=mypy.ini
+
+[testenv:docs]
+description = invoke sphinx-build to build the HTML docs
+basepython = python3
+deps = sphinx >= 7.4, < 8
+# once docstrings are all up-to-date, add the -W flag to turn warnings into errors
+commands =
+    sphinx-build -d "{toxworkdir}/docs_doctree" docs "{toxworkdir}/docs_out" --color -bhtml {posargs}
```

Comment:
You can also set `skip_install = True`, then run tox on the src tree directly. I don't think there is any benefit in running sphinx against the installed source
https://tox.wiki/en/latest/config.html#skip_install

Reply:
Adding `skip_install = True` to `tox.ini`/`[testenv:docs]` means the whole docs build fails because things like `platformdirs` function calls aren't mocked correctly, even after adding practically all the requirements to `conf.py`:`autodoc_mock_imports`.

The ones I've mocked (ftdi, PyDAQmx, PyQt) are because they are platform-dependent / do something fancy at import time.

`CONFIG_DIRECTORY = Path(platformdirs.site_config_dir("Fixate", False))` would be a PITA to mock

Reply:
Ack

Reply:
Can we pin a specific base python here? The reason I went down the path suggested above, was because this failed for me originally. Now, is in some way my own fault because I have python 3.13 beta installed, and tox decided to use that.

If we can pin basepython to 3.12 it would be nice. Especially when we start to build locally and in CI it's nice to have stuff pinned.